OpenRA.Game/Map/Map.cs

```diff
@@ -392,12 +393,16 @@ void PostInit()
 
 			cachedTileSet = Exts.Lazy(() => Rules.TileSets[Tileset]);
 
-			var tl = new MPos(Bounds.Left, Bounds.Top).ToCPos(this);
-			var br = new MPos(Bounds.Right - 1, Bounds.Bottom - 1).ToCPos(this);
-			Cells = new CellRegion(TileShape, tl, br);
+			var tl = new MPos(0, 0).ToCPos(this);
+			var br = new MPos(MapSize.X - 1, MapSize.Y - 1).ToCPos(this);
+			AllCells = new CellRegion(TileShape, tl, br);
+
+			var btl = new MPos(Bounds.Left, Bounds.Top).ToCPos(this);
+			var bbr = new MPos(Bounds.Right - 1, Bounds.Bottom - 1).ToCPos(this);
+			CellsInsideBounds = new CellRegion(TileShape, btl, bbr);
 
 			CustomTerrain = new CellLayer<byte>(this);
-			foreach (var uv in Cells.MapCoords)
+			foreach (var uv in AllCells.MapCoords)
 				CustomTerrain[uv] = byte.MaxValue;
```

Comment:
Resize is specific to the legacy editor, which plays by its own rules. This isn't related to this pr, and that method will be going away soon.